Libraries/LibGUI/BreadcrumbBar.h

```diff
@@ -46,6 +46,7 @@ class BreadcrumbBar : public GUI::Widget {
     Optional<size_t> selected_segment() const { return m_selected_segment; }
 
     Function<void(size_t index)> on_segment_click;
+    Function<void()> on_toggle;
```

Comment:
Let's call this `on_doubleclick` and pass along the `GUI::MouseEvent&` to the callback.
The fact that you want to use it for toggling something is specific to the File Manager changes. :)

Reply:
All good 👍 I was thinking that since the underlying Widget doubleclick_event wasn't public I should somehow further specify its purpose before exposing an event handler, but I see now that my reasoning doesn't exactly make sense 😄